pkg/storage/errors.go

```diff
@@ -68,15 +71,30 @@ func NewUnreachableError(key string, rv int64) *StorageError {
 	}
 }
 
+func NewInvalidUIDError(precondUID *types.UID, objUID types.UID) *StorageError {
+	return &StorageError{
+		Code:       ErrCodeInvalidUID,
+		PrecondUID: precondUID,
+		ObjUID:     objUID,
+	}
+}
+
 type StorageError struct {
 	Code            int
 	Key             string
 	ResourceVersion int64
+	PrecondUID      *types.UID
```

Comment:
I'm not sure if we should explicitly add preconditions here.  In the future we may want to extend them and every change will result in extending this struct with some data that is not needed in 99% of cases.

Instead of that, what I would do is to add:
  "additionalErrorMsg string"
(or sth like that), and would fill it appropriately in case of e.g. invalid uid errors.

WDYT?